### ISSUE TITLE
How to reuse transforms from coco_detection_yolo_format_train for my custom dataset

### ISSUE THREAD

**Issue by hlacikd** [Deci-AI/super-gradients#1677]
### 💡 Your Question

Hello guys,

when i initiate my dataset using 

```
sample_train_data = coco_detection_yolo_format_train(
    dataset_params={
        "data_dir": "/workspaces/rocm-ml/datasets/ds_yolo",
        "images_dir": "train/images",
        "labels_dir": "train/labels",
        "classes": ["lp"],
    },
    dataloader_params={"batch_size": 16, "num_workers": 2},
)
```
i can see that dataset has predefined transforms lists 

```
sample_train_data.dataset.transforms
```

However when i initiate my dataset using YoloDarknetFormatDetectionDataset (which i need due to be able to concat multiple datasets later

```
my_ds = YoloDarknetFormatDetectionDataset(
        data_dir=data_dir, images_dir="train/images", labels_dir="train/labels", input_dim=INPUT_DIM, classes=["lp"], ignore_empty_annotations=True, transforms=train_transforms
    )
```

I have to define my own transforms.

Questions

1. Is there a recipe to load predefined transforms for train/val dataset (the ones that are predefined in dataloaders above?)
2. Is there a recipe to load predefined train_params for yolo-nas, so i do not have to start with creating dictionary (as documented in yolo-nas.md)? So i can only overwrite keys i need.

Thanks!

**Comment by BloodAxe** [comment]
The recipes were designed to act as a way to reproduce a whole training protocol for specific model. That's why they include all dataset params including transformations and other hyperparams. Any main use-case scenario of the recipes is to be used from CLI: `pip -m super_gradients.train_from_recipe --config-name=...`

However sometimes you may want to have Colab notebok to play with. This is where YAML recipes are not that helpful/easy-to-use and this is where you have to instantiate many components by hand, including dataset and transforms.

There is a `load_dataset_params` method that you can use to load a certain dataset params config from recipes. Perhaps you want this config based on the context of the question: https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/recipes/dataset_params/coco_detection_yolo_nas_dataset_params.yaml
Or you can instantiate those transformations by hand. And I actually would recommend doing this way.

Alternatively you can export the whole recipe to self-contained training code using https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/convert_recipe_to_code.py script which you can then put into colab notebook.

**Comment by hlacikd** [comment]
thank you
